### ISSUE TITLE
[Feature Request] Provide python wheels

### ISSUE THREAD

**Issue by sk-** [davisking/dlib#1527]
It would be great if wheels were provided for the [Python package](https://pypi.org/project/dlib), at least the ones for linux (https://github.com/pypa/manylinux), as compilation of the package takes quite a while.

**Comment by davisking** [comment]
Yeah, that would be cool.  Someone should submit a PR :)

**Comment by SmartManoj** [comment]
Why wheels are not compiled for latest version?
https://pypi.org/simple/dlib/#files

**Comment by davisking** [comment]
Because dlib needs to compile itself relative to what hardware is available on you machine. E.g. CUDA, AVX, SSE, NEON, or some BLAS that is available. Those old binaries used a slow lowest common denominator setting and people complain and get confused about why they run slower.

**Comment by josephernest** [comment]
@davisking Are we sure that doing `python setup.py install` automatically uses the best available on the current computer? (i.e. AVX, SSE, NEON if available, etc.)

Or is there there a risk that `python setup.py install` will run a low-spec version whereas a faster version would be available?

TL;DR do we have to do things manually, in addition to `python setup.py install`, to be sure we are using the local machine's features?

**Comment by davisking** [comment]
You should never have to specify this stuff.  The python cmake build checks for what is available and enables everything by default.  If you don't see the other things being enabled it's because the tests for their existence on your computer are coming back false.  The tests should be robust since they build little test programs with those features and run them.  If the programs can build and run then the features are enabled.  If they don't then they aren't enabled.
